lib/ansible/modules/cloud/cloudscale/cloudscale_server.py

```diff
@@ -0,0 +1,438 @@
+        self.wait_for_state('running')
+
+
+    def delete_server(self):
+        self._delete('servers/%s' % self.info['uuid'])
+        self.wait_for_state('absent')
+
+
+    def start_server(self):
+        self._post('servers/%s/start' % self.info['uuid'])
+        self.wait_for_state('running')
+
+
+    def stop_server(self):
+        self._post('servers/%s/stop' % self.info['uuid'])
+        self.wait_for_state('stopped')
+
+
+    def list_servers(self):
+        return self._get('servers')
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec = dict(
+            state               = dict(default='running',
+                                       choices=['running',
+                                                'stopped',
+                                                'absent',
+                                       ]),
+            name                = dict(),
+            uuid                = dict(),
+            flavor              = dict(),
+            image               = dict(),
+            volume_size_gb      = dict(type='int', default=10),
+            bulk_volume_size_gb = dict(type='int'),
+            ssh_keys            = dict(type='list'),
+            use_public_network  = dict(type='bool', default=True),
+            use_private_network = dict(type='bool', default=False),
+            use_ipv6            = dict(type='bool', default=True),
+            anti_affinity_with  = dict(),
+            user_data           = dict(),
+            api_token           = dict(no_log=True),
+        ),
+        required_one_of=(('name', 'uuid'),),
+        mutually_exclusive=(('name', 'uuid'),),
+        supports_check_mode=True,
+    )
+
+    api_token = module.params['api_token'] or os.environ.get('CLOUDSCALE_API_TOKEN')
+
+    if not api_token:
+        module.fail_json(msg='The api_token module parameter or the CLOUDSCALE_API_TOKEN '
+                             'environment varialbe are required for this module.')
+
+    target_state = module.params['state']
+    server = AnsibleCloudscaleServer(module, api_token)
+    current_state = server.info['state']
+
+    if module.check_mode:
+        module.exit_json(changed=not target_state == current_state,
+                         **server.info)
+
+    changed = False
+    if current_state == 'absent' and target_state == 'running':
+        server.create_server()
+        changed = True
+    elif current_state == 'absent' and target_state == 'stopped':
+        server.create_server()
+        server.stop_server()
+        changed = True
+    elif current_state == 'stopped' and target_state == 'running':
+        server.start_server()
+        changed = True
+    elif current_state in ('running', 'stopped') and target_state == 'absent':
+        server.delete_server()
+        changed = True
+    elif current_state == 'running' and target_state == 'stopped':
+        server.stop_server()
+        changed = True
```

Comment:
And in any case I should readd the explcit current_state == target_state check I had at one point because it's better to be safe than sorry. The server could also be paused or in an error state. Don't know how the API presents that.